pkg/cluster/ClusterService.go

```diff
@@ -77,18 +79,22 @@ type ClusterService interface {
 }
 
 type ClusterServiceImpl struct {
-	clusterRepository repository.ClusterRepository
-	logger            *zap.SugaredLogger
-	K8sUtil           *util.K8sUtil
+	clusterRepository  repository.ClusterRepository
+	logger             *zap.SugaredLogger
+	K8sUtil            *util.K8sUtil
+	K8sInformerFactory informer.K8sInformerFactory
 }
 
 func NewClusterServiceImpl(repository repository.ClusterRepository, logger *zap.SugaredLogger,
-	K8sUtil *util.K8sUtil) *ClusterServiceImpl {
-	return &ClusterServiceImpl{
-		clusterRepository: repository,
-		logger:            logger,
-		K8sUtil:           K8sUtil,
-	}
+	K8sUtil *util.K8sUtil, K8sInformerFactory informer.K8sInformerFactory) *ClusterServiceImpl {
+	clusterService := &ClusterServiceImpl{
+		clusterRepository:  repository,
+		logger:             logger,
+		K8sUtil:            K8sUtil,
+		K8sInformerFactory: K8sInformerFactory,
+	}
+	clusterService.buildInformer()
```

Comment:
make it non blocking as we are not dependent on its output

Reply:
its not blocking, any error won't impact.  not dependent on its output

Reply:
done